### PR TITLE
Added workflow for autodocs - images reference

### PR DESCRIPTION
**Pull request by erikaheidi**
This PR creates a workflow to run autodocs via manual trigger, from the GitHub Actions tab, so whenever there's an update to the source images READMEs, a new PR is created here bringing the changes.

We can later on add a schedule to run this periodically. To try it out first, I would prefer to run manually, that's why I didn't add the schedule yet.

Example of PR that will be created: https://github.com/chainguard-dev/deved-autodocs/pull/3

### CODE REVIEW

File: .github/workflows/autodocs-images.yaml
Comment (jamonation) [inline .github/workflows/autodocs-images.yaml:11]:
```suggestion
        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
```

Comment (jamonation) [reply .github/workflows/autodocs-images.yaml:11]: I think there's a preference for using commit hashes with actions, since tags can change invisibly from underneath us. Any reason to use v2 over v3 of this action?

File: .github/workflows/autodocs-images.yaml
Comment (jamonation) [inline .github/workflows/autodocs-images.yaml:16]:
```suggestion
        uses: actions/checkout@@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
```

File: .github/workflows/autodocs-images.yaml
Comment (ltagliaferri) [inline .github/workflows/autodocs-images.yaml:51]: Did you want to add the code owners so this could be a variable?

**Review by ltagliaferri — APPROVED** [review]

File: .github/workflows/autodocs-images.yaml
Comment (jamonation) [inline .github/workflows/autodocs-images.yaml:24]: More curious about this than anything, but could these files get generated in the `deved-autodocs` repo with a GitHub action and uploaded to a cloud storage bucket there, then downloaded as part of this workflow here?

File: .github/workflows/autodocs-images.yaml
Comment (jamonation) [inline .github/workflows/autodocs-images.yaml:25]:
```suggestion
        uses: chainguard-dev/deved-autodocs@84badec14541d344c47d1a8435a8a214e72dbe95 # v1.0.2
```

File: .github/workflows/autodocs-images.yaml
Comment (jamonation) [inline .github/workflows/autodocs-images.yaml:39]:
```suggestion
        uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04 # v4.2.3
```

Comment (jamonation) [reply .github/workflows/autodocs-images.yaml:39]: This action looks like magic! Does it need a github token anywhere to create the PR or does it just figure it out for public repos?

**Review by jamonation — APPROVED** [review]
LGTM! A couple suggestions for versioning things, but they're mostly cosmetic changes.

File: .github/workflows/autodocs-images.yaml
Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml:11]: Ah, thanks! I  have been following mostly the examples they have and I didn't know about using hashes, makes sense. And no reason to use v2, it was just because I copied from another action I had :) thanks!

File: .github/workflows/autodocs-images.yaml
Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml:16]: I will update the version after merging these.

File: .github/workflows/autodocs-images.yaml
Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml:24]: there are some permission challenges for generating the files inside the repo directly, because of the user running in the container image (it usesone of our distroless php images). Uploading would be possible in a separate step since the files are shared

File: .github/workflows/autodocs-images.yaml
Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml:39]: You can set the token, but by default it uses the credentials from the GitHub bot, that's why it shows up as PR author even thou the commit comes from me. One issue we'll have is with the Enforce app, because it won't be gitsigned, but I think there is an action for gitsign that we could use.

File: .github/workflows/autodocs-images.yaml
Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml:51]: Good point, let me try adding that too.

Comment (erikaheidi) [reply .github/workflows/autodocs-images.yaml]: Ok, added the CODEOWNERS file, and removed the reviewers portion since that is not necessary. When this is merged, all PRs will automatically tag @chainguard-dev/deved-team for a review. To enforce the review, I think we'll need to set up branch protection thou.

**Pull request merged**
